### PR TITLE
feat: add `at _` location option to `sym_aggregate`

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

Adds the option to simp-aggregate effects at a specific local hypothesis, using `sym_aggregate at h_foo` as the syntax.

### Testing:

`make_all` ran locally

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tactics/Aggregate.lean [hunk 1]
```diff
@@ -13,22 +13,34 @@ namespace Sym
...
+        let goal ← LNSymSimp (← getMainGoal) ctx simprocs
+        replaceMainGoal goal.toList
+      )
+      (fun _ => pure ())
+
+open Parser.Tactic (location) in
 /--
 `sym_aggregate` will search for all local hypotheses of the form
   `r ?fld ?state = _` or `∀ f ..., r ?fld ?state = _`,
 and use those hypotheses to simplify the goal -/
```
Comment (shigoel) [inline Tactics/Aggregate.lean:42]: Comment needs fixing. `sym_aggregate` can simplify a hyp. too now

File: Tactics/Aggregate.lean [hunk 2]
```diff
@@ -13,22 +13,34 @@ namespace Sym
 
 /-- Given an array of (non-)effects hypotheses, aggregate these effects by
 `simp`ing the main goals -/
-def aggregate (axHyps : Array LocalDecl) : TacticM Unit :=
+def aggregate (axHyps : Array LocalDecl) (location : Location) : TacticM Unit :=
```
Comment (shigoel) [inline Tactics/Aggregate.lean:16]: Comment needs fixing. `aggregate` can also `simp` hyps.

File: Tactics/Aggregate.lean (hunk 1 above)
Comment (alexkeizer) [reply Tactics/Aggregate.lean]: fixed

File: Tactics/Aggregate.lean (hunk 2 above)
Comment (alexkeizer) [reply Tactics/Aggregate.lean:16]: fixed

**Review by shigoel — APPROVED** [review]

**Pull request merged**